### DISCUSSION TITLE
是否有办法获取当前播放内容的缓冲的长度或时间

### DISCUSSION THREAD

**Discussion by GD-Slime** [ywmoyue/biliuwp-lite#619] [General]
最近看到了这篇文章：https://www.bilibili.com/read/cv21206210/
也想将这一功能搬到本软件上来。

获取原始流和减少缓冲时间的功能已经写好了，但是"追帧"这一行为需要获取当前实时的缓冲区长度才能做到自动调整。 
翻了播放器的相关代码也没有找到可以获取当前缓冲区长度或时间的接口或者方法。

现在的播放器是否能做到这个呢? @ywmoyue 

如果不行的话我搞个手动加快一秒进度的也可以。感谢回答！

**Comment by ywmoyue** [comment]
播放器上有没有这个接口这个没了解过，我明天也看看

**Comment by ywmoyue** [comment]
我看FFmpegMediaSource和MediaPlayer都能获取buffer值，追帧跳帧这个我来写吧，之前用flvjs写过类似的

**Reply by GD-Slime** [reply]
好的，感谢！ 我今天把获取原始流和减少缓冲时间的功能最后修一下提个pr上来
